frontend/src/lib/WorkflowParser.test.ts

```diff
@@ -504,6 +504,14 @@ describe('WorkflowParser', () => {
         source: StorageService.GCS,
       });
     });
+
+    it('handles Minio bucket and multi-part key', () => {
```

Comment:
Can you please add more tests here to cover all the use cases just like we do for GCS paths above?